sound/arm/bcm2835-vchiq.c

```diff
@@ -226,7 +226,13 @@ static void audio_vchi_callback(void *param,
 		complete(&instance->msg_avail_comp);
 	} else if (m.type == VC_AUDIO_MSG_TYPE_COMPLETE) {
 		bcm2835_alsa_stream_t *alsa_stream = instance->alsa_stream;
+#if defined(CONFIG_64BIT)
```

Comment:
Yes, it's really a 64 bit issue rather then anything specific to arm.  Now if this driver is going to ever to be used on anything other then arm is another issue.